src/eureka/S5_lightcurve_fitting/models/GPModel.py

```diff
@@ -44,65 +53,74 @@ def __init__(self, kernel_classes, kernel_inputs, lc, gp_code='george',
 
         # Get GP parameters
         self.gp_code_name = gp_code
+        self.normalize = normalize
         self.kernel_types = kernel_classes
         self.kernel_input_names = kernel_inputs
-        self.kernel_input_arrays = []
+        self.kernel_inputs = None
+        self.useHODLR = useHODLR
         self.nkernels = len(kernel_classes)
         self.flux = lc.flux
+        self.fit = np.ones_like(self.flux)
+        self.unc = lc.unc
         self.unc_fit = lc.unc_fit
         self.time = lc.time
 
+        if self.gp_code_name == 'celerite':
+            if self.nkernels > 1:
+                raise AssertionError('Celerite cannot compute multi-'
+                                     'dimensional GPs. Please choose a '
+                                     'different GP code')
+            elif self.kernel_types[0] != 'Matern32':
+                raise AssertionError('Our celerite implementation currently '
+                                     'only supports a Matern32 kernel.')
+
```

Comment:
should we add an error here too for when trying to run tinygp?

Reply:
I didn't since tinygp isn't supported yet anyway, but I also don't see why we would; as far as I can tell, tinygp does support many kernels and multi-dimensional GPs. Or is your point that folks might try to use tinygp even though it isn't supported yet? I'm not too worried about that since the use of tinygp isn't documented anywhere. I also don't know for sure that tinygp wouldn't work if someone managed to install it in their Eureka environment (I wanted to try once but then got distracted).

Reply:
Yeah, I assume it's fine. I just wondered if someone saw it in the code and wanted to try it out and then it won't work and it won't give an error message at all, but I guess it won't matter. From my understanding, I think for it to work we'd have to include jax arrays.